### PR TITLE
features: disable refresh-app-awarness by default again

### PR DESCRIPTION
**Pull request by mvo5**
We enabled the `experimental.refresh-app-awareness` option right
before the 22.04 release [1]. However in hindsight this move was
premature. The UX is not quite ready, the refresh is stopped
but the experience should be that once the app is closed the
update gets applied which is not implemented yet. There are
also design issues with e.g. LXD as outlined in LP:#1978005
the need time and design to resolve. Another reason was the
automatic ~/snap -> ~/Snap folder migration. But this is
done differently now.

For the above reasons this commit changes the default again
to `refresh-app-awarness=false` until there is time to
fix the UX and the LXD issues.

[1] https://github.com/snapcore/snapd/pull/11566

### CODE REVIEW

**Comment by jdstrand** [comment]
IMHO, https://bugs.launchpad.net/snapd/+bug/1975714, particularly https://bugs.launchpad.net/snapd/+bug/1975714/comments/7 is another blocker.

**Comment by mvo5** [comment]
I also asked Gustavo to have a look - mostly for awareness but I feel he should at least be in the loop :)

**Comment by sparkiegeek** [comment]
It would be great to land this (and release in a stable build) ASAP. The bugs[0] break a core tenet of snapd users (that updates occur automatically on a timely basis in the background without intervention).

From the MAAS perspective, we often have snaps in LXD containers for reduced footprint - simply flipping a feature flag back to off-by-default should be low risk for the short term to allow more time and space to address the issues properly. If there's a good use-case for this on e.g. the desktop then perhaps the model or some other piece could set it to true there?

[0] https://bugs.launchpad.net/snapd/+bug/1978834 and others

**Review by mardy — APPROVED** [review]
LGTM, thanks!

File: overlord/hookstate/hooks_test.go [hunk 1]
```diff
@@ -67,12 +67,6 @@ func (s *gateAutoRefreshHookSuite) SetUpTest(c *C) {
 	s.state.Lock()
 	defer s.state.Unlock()
 
-	// disable refresh-app-awareness (it's enabled by default);
-	// specific tests below enable it back.
-	tr := config.NewTransaction(s.state)
-	tr.Set("core", "experimental.refresh-app-awareness", false)
-	tr.Commit()
```
Comment (pedronis) [inline overlord/hookstate/hooks_test.go:74]: do we strictly need to remove this?

**Review by pedronis — APPROVED** [review]
lgtm, one question

File: overlord/hookstate/hooks_test.go (hunk 1 above)
Comment (mvo5) [reply overlord/hookstate/hooks_test.go:74]: Excellent point, I put it back and just updated the comment

**Pull request merged**